tests/unit/test_client.py

```diff
@@ -0,0 +1,481 @@
+from okta.client import Client as OktaClient
+import pytest
+import constants as const
+import yaml
+import os
+
+ERROR_MESSAGE_ORG_URL_MISSING = (
+    "Your Okta URL is missing. You can copy "
+    "your domain from the Okta Developer "
+    "Console. Follow these instructions to"
+    f" find it: {const.FINDING_OKTA_DOMAIN}"
+)
+ERROR_MESSAGE_ORG_URL_NOT_HTTPS = (
+    "Your Okta URL must start with 'https'."
+)
+ERROR_MESSAGE_AUTH_MODE_INVALID = (
+    "The AuthorizationMode configuration "
+    "option must be one of: "
+    "[SSWS, PrivateKey]. "
+    "You provided the SDK with "
+)
+ERROR_MESSAGE_ORG_URL_YOUROKTADOMAIN = (
+    "Replace {{yourOktaDomain}} with your Okta domain. "
+    "You can copy your domain from the Okta Developer Console. "
+    "Follow these instructions to find it: "
+    f"{const.FINDING_OKTA_DOMAIN}"
+)
+
+ERROR_MESSAGE_ORG_URL_ADMIN = (
+    "Your Okta domain should not contain -admin. "
+)
+
+ERROR_MESSAGE_ORG_URL_TYPO = (
+    "It looks like there's a typo in your Okta domain."
+)
+
+ERROR_MESSAGE_API_TOKEN_MISSING = (
+    "Your Okta API token is missing. "
+    "You can generate one in the Okta Developer"
+    " Console. Follow these instructions:"
+    f" {const.GET_OKTA_API_TOKEN}"
+)
+
+ERROR_MESSAGE_API_TOKEN_DEFAULT = (
+    "Replace {{apiToken}} with your Okta API token. "
+    "You can generate one in the Okta Developer Console. "
+    f"Follow these instructions: {const.GET_OKTA_API_TOKEN}"
+)
+
+ERROR_MESSAGE_CLIENT_ID_MISSING = (
+    "Your client ID is missing. You can copy it from the "
+    "Okta Developer Console in the details for the Application "
+    "you created. Follow these instructions to find it: "
+    f"{const.FINDING_OKTA_APP_CRED}"
+)
+
+ERROR_MESSAGE_CLIENT_ID_DEFAULT = (
+    "Replace {{clientId}} with the client ID of your Application. "
+    "You can copy it from the Okta Developer Console in the "
+    "details for the Application you created. Follow these "
+    f"instructions to find it: {const.FINDING_OKTA_APP_CRED}"
+)
+
+ERROR_MESSAGE_SCOPES_PK_MISSING = (
+    "When using authorization mode 'PrivateKey', you must supply "
+    "'okta.client.scopes' and 'okta.client.privateKey'"
+)
```

Comment:
I like the use of constants here, but I wonder if we should move these out of the test itself and have a single source for these.  They are defined again in the config validation, so maybe we can define them somewhere else central?